pager/src/main/java/com/google/accompanist/pager/PagerState.kt

```diff
@@ -436,3 +441,21 @@ class PagerState(
         )
     }
 }
+
+/**
+ * A flow which emits the [PagerState.currentPage] if it changes due a scroll
+ * has completed.
+ *
+ * This flow is not meant to be used for updating any UI within the attached [HorizontalPager]
+ * or [VerticalPager]. For that use-case, you should read
+ * [PagerScope.currentPage] and [PagerScope.currentPageOffset] from the content scope.
+ *
+ * @sample com.google.accompanist.sample.pager.PageChangedFlowSample
+ */
+@ExperimentalPagerApi
+inline val PagerState.pageChangedFlow: Flow<Int>
+    get() = snapshotFlow { isScrollInProgress }
+        // Only emit when the scroll has finished
+        .filter { isScrollInProgress -> !isScrollInProgress }
+        .map { currentPage }
+        .distinctUntilChanged()
```

Comment:
Is this needed? Shouldn't the filter enforce that emissions only get here when scrolling finishes?

Reply:
Not necessarily. A scroll could finish and the pager snaps back to the original page.

Reply:
True, true.

Reply:
How about making this a StateFlow with stateIn? You wouldn't need the distinctUntilChanged and since this is a PagerState it sounds appropriate (but I don't know the details of how it's consumed)

Reply:
Happy to change it, but not sure what the benefit would be? Would the type be `StateFlow` or `Flow`?